walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,101 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.ListIterator;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  // comparator using the Java 8 lambda syntax
+  private final Comparator<Event> ORDER_END =
+      (Event a, Event b) -> Long.compare(a.getWhen().end(), b.getWhen().end());
+
+  public Collection<TimeRange> query(Collection<Event> events,
+                                     MeetingRequest request) {
+    /* Algorithm overview:
+    Remove people from the event list who are in the list but not in the
+    meeting. Sort all of events in descending order by end time. Grab the start
+    time for this current event and traverse through array all events that have
+    an end time greater than this start time. The time slot that can not be used
+    is then the end time of the current iteration and the minimum start time of
+    those other events. Now take the event that after the last event that was
+    discovered still within the constraints and repeat from there. Such an
+    approach will take O(nlogn) time to sort and O(n) time to iterate. The
```

Comment:
I like this overview, but I think it's a little too detailed. It will be easier to read this function if you make the overview just a couple sentences, then distribute the more detailed comments to the relevant parts of the code. Otherwise, the reader may get bogged down in mental arithmetic (current, next, end time greater than start, but we're going backwards...) before even seeing the code.